projects/pgai/tests/vectorizer/test_vectorizer_cli.py

```diff
@@ -0,0 +1,420 @@
+import logging
+from collections.abc import Generator
+from typing import Any
+
+import openai
+import psycopg
+import pytest
+from click.testing import CliRunner
+from psycopg import sql
+from psycopg.rows import dict_row
+from testcontainers.postgres import PostgresContainer  # type: ignore
+
+from pgai.cli import vectorizer_worker
+from tests.vectorizer import expected
+
+
+@pytest.fixture(scope="session")
+def cli_postgres_container():
+    """Provides a clean Postgres container for CLI testing"""
+    with PostgresContainer("timescale/timescaledb-ha:pg16", driver=None) as postgres:
+        yield postgres
+
+
+@pytest.fixture
+def cli_db(
+    cli_postgres_container: PostgresContainer,
+) -> Generator[dict[str, Any], None, None]:
+    """Creates a test database with pgai installed"""
+    role = "tsdbquerier"
+    password = "my-password"
+    db_host = cli_postgres_container._docker.host()  # type: ignore
+
+    # Connect and setup initial database
+    with psycopg.connect(
+        cli_postgres_container.get_connection_url(host=db_host),
+        autocommit=True,
+    ) as conn:
+        # Setup test user
+        conn.execute(sql.SQL("DROP USER IF EXISTS {}").format(sql.Identifier(role)))
+        conn.execute(
+            sql.SQL("CREATE USER {} WITH SUPERUSER PASSWORD {}").format(
+                sql.Identifier(role), sql.Literal(password)
+            )
+        )
+
+        # Install pgai
+        conn.execute("CREATE EXTENSION IF NOT EXISTS ai CASCADE")
+
+        yield {
+            "container": cli_postgres_container,
+            "conn": conn,
+            "event_db_config": {
+                "host": db_host,
+                "port": int(
+                    cli_postgres_container.get_exposed_port(cli_postgres_container.port)
+                ),
+                "db_name": cli_postgres_container.dbname,
+                "ssl_mode": "disable",
+                "role": role,
+                "password": password,
+            },
+        }
+
+
+@pytest.fixture
+def cli_db_url(cli_db: dict[str, Any]) -> str:
+    """Constructs database URL from the cli_db fixture"""
+    config = cli_db["event_db_config"]
+    return f"postgres://{config['role']}:{config['password']}@{config['host']}:{config['port']}/{config['db_name']}?sslmode={config['ssl_mode']}"
```

Comment:
As mentioned above, this could become:
```suggestion
    return cli_db["container"].get_connection_url()
```

Or (if we want to keep the custom "host" thing which I don't understand):

```suggestion
    container = cli_db["container"]
    return container.get_connection_url(container.get_docker_client().host())
```